File: lua/gtk4/action-bar.lua
local lgi = require("lgi")
local Gtk = lgi.require("Gtk", "4.0")
local Gio = lgi.Gio

-- GtkActionBar: A full width container to add contextual actions

local appID = "io.github.Miqueas.GTK-Examples.Lua.Gtk4.ActionBar"
local appTitle = "GtkActionBar"
local app = Gtk.Application.new(appID, Gio.ApplicationFlags.FLAGS_NONE)

function app:on_startup()
  local window = Gtk.ApplicationWindow.new(self)
  local titleLabel = Gtk.Label.new(([[<span font="bold">%s</span>]]):format(appTitle))
  local headerBar = Gtk.HeaderBar.new()
  local actionBar = Gtk.ActionBar.new()
  local content = Gtk.Label.new("Here is the content of your app")
  local mainBox = Gtk.Box()

  self:add_window(window)

  window.child = mainBox
  window:set_titlebar(headerBar)
  window:set_default_size(400, 400)

  titleLabel.use_markup = true

  headerBar.title_widget = titleLabel
  headerBar.show_title_buttons = true

  actionBar:pack_start(Gtk.Label.new("Something"))
  actionBar:pack_end(Gtk.Button.new_with_label("A button"))

  content.vexpand = true

  mainBox.orientation = Gtk.Orientation.VERTICAL
  mainBox:append(content)
  mainBox:append(actionBar)
end

function app:on_activate()
  self.active_window:present()
end

return app:run(arg)

File: python/gtk4/action-bar.py
import gi, sys
gi.require_version("Gtk", "4.0")
from gi.repository import Gtk

# GtkActionBar: A full width container to add contextual actions

appID = "io.github.Miqueas.GTK-Examples.Python.Gtk4.ActionBar"
appTitle = "Gtk.ActionBar"

def onAppStartup(self):
    global appTitle

    window = Gtk.ApplicationWindow(application = self, title = appTitle)
    box = Gtk.Box(orientation = Gtk.Orientation.VERTICAL)
    label = Gtk.Label(label = "App content here", vexpand = True, hexpand = True)
    actionBar = Gtk.ActionBar()

    window.set_child(box)
    window.set_default_size(400, 400)

    actionBar.pack_start(Gtk.Label(label = "Something"))
    actionBar.pack_end(Gtk.Button(label = "A button"))

    box.append(label)
    box.append(actionBar)

app = Gtk.Application(application_id = appID, flags = 0)
app.connect("activate", lambda self: self.get_active_window().present())
app.connect("startup", onAppStartup)

sys.exit(app.run(sys.argv))